### PR TITLE
Refactor to move large lists to simple files

### PR DESCRIPTION
**Pull request by antgonza**
Back story: If you have a dataset that is really diverse it's possible that you will get hits to a lot of different sequences. Thus, the variable that counts how many times a sequence has hit grows and can reach to really high memory usage - also making it slow as it had to search for the sequence name.

In this PR I'm removing the memory footprint by moving one of the results options to simple files, where the every hit header gets stored. This, will allow the user to simply do `cat file | sort | uniq -c` to get a summary. 

The other part that is being removed is the database check to see if a hit is present in both the interest and rest db. In the past, we have thought that this should be removed and perhaps checked in other step so is fine to remove.

### CODE REVIEW

**Comment by ElDeveloper** [comment]
@antgonza, would you mind pulling from upstream and pushing back up? The diff is a bit confusing as it includes the changes from #40

**Comment by antgonza** [comment]
Not that the wrong check should be fine, it's because the coverage decreased: 0.02% !

**Comment by ElDeveloper** [comment]
Thanks, having a look at it just now.

File: platypus/commands.py [hunk 1]
```diff
@@ -50,10 +49,10 @@ def compare(interest_fp, other_fp, output_dir='blast-results-compare',
         other database search results. If None is passed, it defaults to
         `[50]`.
     hits_to_first : bool, optional defaults to False
-        Outputs the labels and counts of the sequences being hit in the first
+        Outputs all the labels of the sequences being hit in the first
```
Comment (ElDeveloper) [inline platypus/commands.py]: Can you update this in `scripts/platypus`, specifically in descriptions in lines 56 and 59.

**Comment by ElDeveloper** [comment]
Thanks @antgonza, just one comment. In addition to that, would you mind adding a note in the ChangeLog.md file, with a brief user-level description of the following text:

> In this PR I'm removing the memory footprint by moving one of the results options to simple files, where the every hit header gets stored. This, will allow the user to simply do cat file | sort | uniq -c to get a summary.

Finally, are there any updates that need to be made to TUTORIAL.md?

**Comment by antgonza** [comment]
Comments have been addressed. Note that the TUTORIAL.md doesn't need to be updated.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,9 @@ Platypus Conquistador 0.9.0-dev (changes since Platypus Conquistador 0.9.0 go he
 -----------------------------------------------------------------------------------
 
 - Changed parse.parse_m9 so it doesn't load the full file into memory
+- Memory requirements was improved by moving the labels and counts of hits to the databases
```
Comment (ElDeveloper) [inline CHANGELOG.md]: `Memory requirements was improved` -> `Improved memory requirements`?

**Comment by ElDeveloper** [comment]
Thanks!

**Pull request merged**